llvm/lib/IR/RepoHashCalculator.cpp

```diff
@@ -337,9 +337,18 @@ void HashCalculator::hashGlobalValue(const GlobalValue *V) {
   }
 
   if (auto *GO = dyn_cast<GlobalObject>(V)) {
-    // Push GO into the dependent list if it is not a declaration.
-    if (!GO->isDeclaration())
-      getDependencies().emplace_back(GO);
+    // Don't push GO into the dependent list if it is a declaration.
+    if (GO->isDeclaration())
+      return;
+
+    // Don't push GO into the dependent list if function will not be inlined
+    // and not be discarded if it is not used.
+    const llvm::Function *const Fn = dyn_cast<const llvm::Function>(GO);
+    if (Fn && Fn->hasFnAttribute(Attribute::NoInline) &&
+        !Fn->isDiscardableIfUnused())
+      return;
```

Comment:
This seems to be an unrelated change. Is there a connection to the dependencies/contributions work?

Reply:
Yes, it is unrelated change to improve the performance. Should I create a new PR for code review?

Reply:
Given that this PR is relatively complex, yes that might be a good idea.

Reply:
Thanks Paul. I have created a new PR #114 for performance improvement.